build/test/httpserver_test.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+# Test the Easegress' basic functionality which is generating
+# an HTTPServer and Pipeline for testing HTTP Requests.
+# author:   benjaminwu 
+# date:     2021/0716
+
+set -e
+
+# path related define.
+# Note: use $(dirname $(realpath ${BASH_SOURCE[0]})) to value SCRIPTPATH is OK in linux platform, 
+#       but not in MacOS.(cause there is note `realpath` in it)
+# reference: https://stackoverflow.com/questions/4774054/reliable-way-for-a-bash-script-to-get-the-full-path-to-itself
+SCRIPTPATH="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+pushd $SCRIPTPATH"/../../example" > /dev/null
+EXAMPLEDIR="$SCRIPTPATH"/../../example
+WRITER01DIR=$EXAMPLEDIR"/writer-001"
+
+# target file related define.
+server="writer-001/bin/easegress-server"
+backend="$EXAMPLEDIR/backend-service/mirror/mirror.go"
+
+# color define.
+COLOR_NONE='\033[0m'
+COLOR_INFO='\033[0;36m'
+COLOR_ERROR='\033[1;31m'
+
+# clean cleans writer-001's cluster data and the `go run` process.
+function clean()
+{
+     # basic cleaning routine
+     bash $EXAMPLEDIR/stop_cluster.sh 
+     bash $EXAMPLEDIR/clean_cluster.sh
+
+
+     # clean the go mirror backend
+     if [ "$1" != "" ];then
+        echo -e "\n${COLOR_INFO}finish mirror running pid=$1${COLOR_NONE}"
+	child_pid=`pgrep -P $1`
+
+	if [ "$child_pid" != "" ]; then
+	    kill -9 $child_pid
+            echo -e "\n${COLOR_INFO}finish mirror running child process pid=$child_pid${COLOR_NONE}"
+	fi
+
+	kill -9 $1
+     fi
+
+}
+
+# clean the cluster resource first.
+clean
+
+# start writer01 for testing. 
+start_svr=`$WRITER01DIR/start.sh `
+
+# wait Easegress to be ready
+sleep 2
```

Comment:
can we use a better way to do this? such as https://unix.stackexchange.com/questions/5277/how-do-i-tell-a-script-to-wait-for-a-process-to-start-accepting-requests-on-a-po

Reply:
Replacing it with `while ! nc -z localhost $mirror_port</dev/null; do sleep 3; done`.